### PR TITLE
[KIECLOUD-145] add replicas option

### PR DESCRIPTION
**Pull request by tchughesiv**
- added replicas option for console, server, and smartrouter objects
 - reconciliation occurs for all DC objects to ensure proper replicas set (including amq and dbs)
 - console replica scaling only allowed in HA envs and PAM prod envs
 - fixed server-config example
 - changed go template delimiters to `[[ ]]` in order to allow for integer subs (yaml formatting related)
```yaml
        replicas: [[.Console.Replicas]]
```

/assign @bmozaffa @ruromero 

Signed-off-by: tchughesiv <tchughesiv@gmail.com>

### CODE REVIEW

File: pkg/controller/kieapp/defaults/defaults.go [hunk 1]
```diff
@@ -120,9 +132,35 @@ func getSmartRouterTemplate(cr *v1.KieApp) v1.SmartRouterTemplate {
 		template.KeystoreSecret = cr.Spec.Objects.SmartRouter.KeystoreSecret
 	}
 
+	// Set replicas
+	var denyScale bool
```
Comment (ruromero) [inline pkg/controller/kieapp/defaults/defaults.go]: Extract this into a method

File: pkg/controller/kieapp/defaults/defaults.go [hunk 2]
```diff
@@ -105,6 +106,17 @@ func getConsoleTemplate(cr *v1.KieApp) v1.ConsoleTemplate {
 	} else {
 		template.KeystoreSecret = cr.Spec.Objects.Console.KeystoreSecret
 	}
+	// Set replicas
+	var denyScale bool
```
Comment (ruromero) [inline pkg/controller/kieapp/defaults/defaults.go]: Extract this into a method, duplicated code with L136:L144

File: pkg/controller/kieapp/defaults/defaults.go [hunk 3]
```diff
@@ -170,32 +208,44 @@ func getServersConfig(cr *v1.KieApp, commonConfig *v1.CommonConfig) ([]v1.Server
...
 				Name:      fmt.Sprintf("%s-kieserver:latest", commonConfig.ApplicationName),
 				Namespace: "",
 			}
 		} else {
-			crTemplate.From = getDefaultKieServerImage(commonConfig, serverSet.From)
+			template.From = getDefaultKieServerImage(commonConfig, serverSet.From)
+		}
+
+		// Set replicas
+		var denyScale bool
```
Comment (ruromero) [inline pkg/controller/kieapp/defaults/defaults.go]: Same code as before, should be moved to a new method

File: pkg/controller/kieapp/kieapp_controller_test.go [hunk 4]
```diff
@@ -48,7 +47,7 @@ func TestUnknownEnvironmentObjects(t *testing.T) {
 	env, err := defaults.GetEnvironment(cr, test.MockService())
 	assert.Equal(t, fmt.Sprintf("envs/%s.yaml does not exist, '%s' KieApp not deployed", cr.Spec.Environment, cr.Name), err.Error())
 
-	env = consolidateObjects(env, cr)
+	env = defaults.ConsolidateObjects(env, cr)
```
Comment (ruromero) [inline pkg/controller/kieapp/kieapp_controller_test.go]: Move to defaults_test.go. It is only testing the ConsolidateObjects function.
Applies to
* TestUnknownEnvironmentObjects
* TestTrialConsoleEnv
* TestTrialServerEnv
* TestTrialServersEnv

**Comment by tchughesiv** [comment]
@ruromero changes complete

**Pull request merged**